tests/unit/test_space.py

```diff
@@ -308,6 +308,20 @@ def test_box_sampling(num_samples: int) -> None:
     _assert_correct_number_of_unique_constrained_samples(num_samples, box, samples)
 
 
+@pytest.mark.parametrize("num_samples", [0, 1, 10])
+def test_box_sobol_sampling(num_samples: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    sobol_samples = box.sample_sobol(num_samples)
+    _assert_correct_number_of_unique_constrained_samples(num_samples, box, sobol_samples)
+
+
+@pytest.mark.parametrize("num_samples", [0, 1, 10])
+def test_box_halton_sampling(num_samples: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    halton_samples = box.sample_halton(num_samples)
+    _assert_correct_number_of_unique_constrained_samples(num_samples, box, halton_samples)
+
+
```

Comment:
Please also add two tests to check that halton/sobol return the same sequence when given the same random seed / skip. For example, see line 342 in https://github.com/secondmind-labs/trieste/blob/develop/tests/unit/acquisition/test_sampler.py

Reply:
Do you mean to test if two methods should return different sequences?

Reply:
test that halton returns the same ouptut for a given random seed and different outputs for different seeds. And also check the same for Sobol.

Reply:
I see okayy 👌🏼

Reply:
for Halton the seed check doesn't pass, then I checked on the python terminal if the problem is in the tfp and got: 
```
>>> tfp.mcmc.sample_halton_sequence(dim=2,num_results=5,seed=1)
<tf.Tensor: shape=(5, 2), dtype=float32, numpy=
array([[0.30109066, 0.61624485],
       [0.55109066, 0.94957817],
       [0.05109064, 0.06068926],
       [0.92609066, 0.3940226 ],
       [0.42609066, 0.72735596]], dtype=float32)>
>>> tfp.mcmc.sample_halton_sequence(dim=2,num_results=5,seed=1)
<tf.Tensor: shape=(5, 2), dtype=float32, numpy=
array([[0.28878212, 0.07074466],
       [0.5387821 , 0.7374114 ],
       [0.0387821 , 0.5151891 ],
       [0.9137821 , 0.18185577],
       [0.41378212, 0.8485224 ]], dtype=float32)>
```
I'll investigate it more

Reply:
Cheers

Reply:
as you can see above using the same seed the tfp's halton sample returns the different points. tried to set tf random seed still got the same behavior. do you think there is a bug on the tfp.mcmc.sample_halton_sequence?

Reply:
I'm not sure. Maybe raise an issue or check with the tfp people?

Reply:
I'll make an issue on tfp github

Reply:
okay, so we need to set and reset `tf.random.set_seed()` with the same seed in between calls to make it works.
thanks to @TsingQAQ  who pointed this out to me.